rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,143 @@
+
+Recently, we have carried out some proof of concept testing of the Carbon components by integrating Applitools with 18 component test feature files from our existing Cypress regression suite. Steps were added into the feature files and relevant support files to enable Applitools to take a snapshot after each test scenario. While we can see that this would be a more accurate way of testing the visual attributes of the components, this activity has highlighted that in its current form, the regression suite with Applitools integration is not a viable option as the vast number of comparisons extend an already lengthy suite. 
+
+## Reformatting of component stories for testing
+By presenting the various different pre-set configurations of the components in as few preview frames and stories as possible without the need to set knobs, the tests could run faster while Applitools could take and compare a vastly reduced number of snapshots. This will not impact the ability of our library to showcase to Developers how they can create a beautiful UI in a fast and easy manner with our components.
+
+# Detailed design
+
+To facilitate a change to this method of regression testing, the components will need to be presented differently in Storybook in `.mdx` files with enough possible configurations pre-set in the stories in the `Design System` directory or main component list with few to no knobs to amend to showcase the components. Events should be tested from here where possible. Any further configurations required to completely test the component will be contained in the `Test` directory. This has already been done to some extent for some components e.g. Button in the `Design System` directory
+
+![CARBON](./../../.assets/button_design_system.png)
+
+and the `All Buttons` story for testing purposes
+
+![CARBON](./../../.assets/all_buttons_story.png)
+
+Existing functional or event tests which Cypress is required for will need to be amended to test in the `Docs` tab of the new refactored stories rather than in the `Canvas`. The accessibility tests should not be affected. Any visual verification tests currently done by Cypress alone should be done by Applitools. Any additional cases, such as edge cases, cases that cannot be tested in the component in the `Design System` directory or extended test cases e.g. tests for special characters in labels, should be added in an additional story in the `Test` directory. If all of the configurations can be demonstrated in a single story in the `Test` directory then they should, with visual regression testing run against those stories. We should, however, take care to test only what we deem as bringing value to our regression testing and not have Applitools snapshot everything just because we can.
```

Comment:
e.g. tests for special characters in labels - here should be also mentioned other tests we should perform such as: keyboard accessibility, action events tests

Reply:
Didn't we agree that action events tests could be done from the stories in `Design System` where possible as those stories would be changed to match what was done in one component recently to allow that? Also, I thought the accessibility tests would continue to be done where they currently are now, with most of the components being tested from `Design System` or the main component list? I thought we'd agreed that only those tests that could not be done from `Design System` and additional specific visual testing would be done in the `Test` directory?